mozilla_aws_cli/cli.py

```diff
@@ -79,17 +83,24 @@ def validate_config_file(ctx, param, filenames):
         # Override the --config file contents with the mozilla_aws_cli_config
         # module contents
         for key in mozilla_aws_cli_config.config:
-            config.set('DEFAULT', key, mozilla_aws_cli_config.config[key])
+            if key in config.defaults() and config.defaults()[key] != mozilla_aws_cli_config.config[key]:
+                raise click.BadOptionUsage(None, "setting for `{}` exists in both global module and local config file".format(key))
+
+            config.defaults()[key] = mozilla_aws_cli_config.config[key]
 
     missing_settings = (
         {'client_id', 'idtoken_for_roles_url', 'well_known_url'} - set(config.defaults().keys()))
 
     if missing_settings:
-        raise click.BadParameter(
-            '{} setting{} are missing from the config files {}'.format(
-                ', '.join(missing_settings),
+        message = '{} setting{} are missing from config files: {}'.format(
+                ', '.join(["`{}`".format(setting) for setting in missing_settings]),
                 's' if len(missing_settings) > 1 else '',
-                " ".join(filenames)))
+                " ".join(filenames))
```

Comment:
This was hard for me to understand. Here's the same thing in a more [explicit](https://www.python.org/dev/peps/pep-0020/) form.

```
        missing_setting_list = ', '.join(["`{}`".format(setting) for setting in missing_settings])
        plural = 's are' if len(missing_settings) > 1 else ' is'
        filename_list = " ".join(filenames)
        message = '{missing_setting_list} setting{plural} missing from config files: {filename_list}'.format(
            missing_setting_list=missing_setting_list,
            plural=plural,
            filename_list=filename_list)
```